framework/src/org/checkerframework/framework/type/AnnotatedTypeLoader.java

```diff
@@ -0,0 +1,372 @@
+    // retrieve its resource URL (either a jar or regular file directory), then
+    // proceeds to reflectively load annotations if it is valid. Returns an
+    // empty set if it isn't a valid URL
+    private void ProcessPackageNameAndLoadAnnos() {
+        //resourceURL = Thread.currentThread().getContextClassLoader().getResource(packageName.replace(DOT, SLASH));
+        resourceURL = this.getClass().getClassLoader().getResource(packageName.replace(DOT, SLASH));
+
+        // in checkers, there will be a resource URL for the qual directory. But
+        // when called in the framework (eg GeneralAnnotatedTypeFactory), there
+        // won't be a resourceURL since there isn't a qual directory.
+        if (resourceURL != null) {
+            // if there's a URL, load from it
+            loadedAnnotations = loadCheckerFrameworkAnnotations();
+        } else {
+            // otherwise create an empty set
+            loadedAnnotations = new HashMap<Class<? extends Annotation>, AnnotationMirror>();
+        }
+    }
+
+    /**
+     * Every subclass of AnnotatedTypeLoader must implement how it will convert a class into an annotation mirror
+     * @param annoClass an annotation's class
+     * @return AnnotationMirror of the annotation, or null if this annotation isn't required
+     */
+    protected AnnotationMirror createAnnotationMirrorFromClass(Class<? extends Annotation> annoClass) {
+        // build the annotation mirror
+        AnnotationBuilder builder = new AnnotationBuilder(processingEnv, annoClass);
+        AnnotationMirror annoMirroResult = builder.build();
+        return annoMirroResult;
+    }
+
+    /**
+     * gets the qualifier set from the loaded annotations, with option to skip {@link PolyAll}
+     * @param excludePolyAll setting this to true would skip adding {@link PolyAll} to the list of qualifiers
+     * @return an unmodifiable set of the qualifiers
+     */
+    public final Set<Class<? extends Annotation>> getAnnotatedTypeQualifierSet(final boolean excludePolyAll) {
+        Set<Class<? extends Annotation>> annotatedTypeQualSet = new HashSet<Class<? extends Annotation>>();
+        annotatedTypeQualSet.addAll(loadedAnnotations.keySet());
+
+        if (!excludePolyAll) {
+            // add PolyAll to the qualifier set unless it is explicitly excluded
+            annotatedTypeQualSet.add(PolyAll.class);
+        }
+
+        // return the set of annotation classes
+        return Collections.unmodifiableSet(annotatedTypeQualSet);
+    }
+
+    /**
+     * gets the set of annotation mirrors from the loaded annotations
+     * @return a hashset of the annotation mirrors
+     */
+    public final Set<AnnotationMirror> getAnnotationMirrorSet() {
+        return new HashSet<AnnotationMirror>(loadedAnnotations.values());
+    }
+
+    // loads annotations via reflection
+    private final Map<Class<? extends Annotation>, AnnotationMirror> loadCheckerFrameworkAnnotations() {
```

Comment:
I think this should be loadTypeQualifiers or loadTypeAnnotations. It's not clear what a "Checker Framework Annotation" is.